### DISCUSSION TITLE
Reverse Proxy with nginx does not seem to be working! Where am I going wrong?

### DISCUSSION THREAD

**Discussion by cypht** [koush/scrypted#945] [Q&A] [answered]
Hello,

I have scrypted and nginx running is separate containers. Trying to reverse proxy to scrypted but it just gives a 404 error. What am I doing wrong?

server {
        server_name scrypted.gsdhm.net;
        listen 80;
        access_log /var/log/nginx/access.log;
        return 301 https://$host$request_uri;
}
server {
    listen  443 ssl http2;
    listen  [::]:443 ssl http2;
    server_name scrypted.gsdhm.net;

    include /etc/nginx/ssl/*;

    location / {
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Host $server_name;
        proxy_set_header Host $http_host;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-Ssl on;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_pass https://172.168.22.65:10443;
    }
}

**Comment by koush** [comment, marked as answer]
```


map $http_upgrade  $upgradevalue {
    websocket        websocket;
    default  '';
}
map $http_upgrade  $connectionvalue {
    websocket        Upgrade;
    default  '';
}

server {
    # Listen to port 443 on both IPv4 and IPv6.
    listen 443 ssl;
    listen [::]:443 ssl;

    # Domain names this server should respond to.
    server_name scrypted.example.com;

    # Load the certificate files.
    ssl_certificate /etc/letsencrypt/live/scrypted.example.com/fullchain.pem;
    ssl_certificate_key /etc/letsencrypt/live/scrypted.example.com/privkey.pem;
    ssl_trusted_certificate /etc/letsencrypt/live/scrypted.example.com/chain.pem;

    # Load the Diffie-Hellman parameter.
    ssl_dhparam /etc/letsencrypt/dhparams/dhparam.pem;

    location / {
        proxy_pass https://192.168.2.124:52960/;
        proxy_ssl_verify              off;

        proxy_set_header Upgrade $upgradevalue;
        proxy_set_header Connection $connectionvalue;
        proxy_set_header Host "scrypted.example.com";
        proxy_http_version 1.1;
        proxy_buffering off;
    }
}
```

**Comment by cypht** [comment]
That worked. Perfect. Thank You!!
